spec/models/order_updater_spec.rb

```diff
@@ -101,49 +101,30 @@
 
   context '#before_save_hook' do
     let(:distributor) { build(:distributor_enterprise) }
-    let(:order) { build(:order, distributor: distributor) }
-    let(:shipment) { build(:shipment) }
-    let(:shipping_rate) do
-      Spree::ShippingRate.new(
-        shipping_method: shipping_method,
-        selected: true
-      )
-    end
+    let(:shipment) { create(:shipment_with, :shipping_method, shipping_method: shipping_method) }
```

Comment:
Isn't the use of the `:shipping_method` trait a bit redundant here? As we're passing a shipping method in this would be equivalent of the simpler `create(:shipment, shipping_method: shipping_method)`. Am I missing something?

Reply:
I dont think so.
The spree shipment factory does create(:shipping_method)
but the new ofn shipment_with factory has the same create(:shipping_method) but inside a transient block, which means that if shipping_method is provided, it will get used.
and in this case we are providing different shipping_methods, once as pickup and then as delivery.
Makes sense?

Reply:
So, for me to understand, if no shipping method is provided none will be created, am I right?

Reply:
no, if no shipping_method is provided the code inside transient is used, in this specific shipment_with factory, the default :shipping_method factory will be used.